### PR TITLE
fix: use memoized fn in use-interval (#2029)

### PR DESCRIPTION
**Pull request by kongmoumou**
[[中文版模板 / Chinese template](https://github.com/alibaba/hooks/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

#2029 

### 💡 Background and solution

使用 useMemoizedFn 替换 useLatest 解决与 devtool 冲突的问题

详见 https://github.com/alibaba/hooks/issues/728#issuecomment-1200118142

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by kongmoumou** [comment]
@li-jia-nan could u make a review 😂

**Review by li-jia-nan — APPROVED** [review]

**Comment by li-jia-nan** [comment]
感觉命名可以再优化一下，cb 有点敷衍了

**Comment by liuyib** [comment]
cb -> memoizedFn      :)

**Comment by kongmoumou** [comment]
@li-jia-nan @liuyib done

**Comment by li-jia-nan** [comment]
- #2092 

useTimeout 在 react devtools 中也有问题，顺便一起修了

File: packages/hooks/src/useInterval/index.ts [hunk 1]
```diff
@@ -11,25 +11,23 @@ function useInterval(
...
-      fnRef.current();
-    }, delay);
+    timerRef.current = setInterval(timerCallback, delay);
     return () => {
       if (timerRef.current) {
         clearInterval(timerRef.current);
       }
     };
-  }, [delay]);
+  }, [delay, timerCallback]);
```
Comment (liuyib) [inline packages/hooks/src/useInterval/index.ts:30]: `timerCallback` 放在依赖数组里没有意义，它是个 ref 值，不用放在依赖数组里。我觉着这种无意义的写法还是要避免下

Comment (liuyib) [reply packages/hooks/src/useInterval/index.ts:30]: @li-jia-nan  @crazylxr  你们觉着呢？

Comment (Wxh16144) [reply packages/hooks/src/useInterval/index.ts:30]: > timerCallback 放在依赖数组里没有意义

疑惑+1，为啥被折叠了？

Comment (li-jia-nan) [reply packages/hooks/src/useInterval/index.ts:30]: 确实没意义，删了吧

**Pull request merged**